clients/baseclient/client_util.go

```diff
@@ -23,30 +22,18 @@ func shouldRetry(err error) bool {
 	if err == nil {
 		return false
 	}
-	if isMatching(err) {
```

Comment:
why is this check no longer needed?

Reply:
I think this check is no longer needed because if the check for nil error is passed, it's enough to just check the status code and act accordingly. If I'm mistaken I'll bring the check back. What are your thoughts on the matter ?